src/test/Components/CreateImageWizardV2/CreateImageWizard.azure.test.js

```diff
@@ -0,0 +1,285 @@
+
+    await user.click(
+      screen.getByRole('radio', {
+        name: /use an account configured from sources\./i,
+      })
+    );
+
+    await waitFor(() => expect(nextButton).toHaveClass('pf-m-disabled'));
+
+    const sourceDropdown = await getSourceDropdown();
+
+    // manual values should be cleared out
+    expect(
+      screen.getByRole('textbox', {
+        name: /azure tenant guid/i,
+      })
+    ).toHaveValue('');
+
+    expect(
+      screen.getByRole('textbox', {
+        name: /subscription id/i,
+      })
+    ).toHaveValue('f8f200aa-6234-4bfb-86c2-163d33dffc0c');
+
+    await user.click(sourceDropdown);
+
+    await user.click(
+      await screen.findByRole('option', {
+        name: /azureSource1/i,
+      })
+    );
+    // wait for fetching the upload info
+    await waitFor(() =>
+      expect(
+        screen.getByRole('textbox', {
+          name: /azure tenant guid/i,
+        })
+      ).not.toHaveValue('')
+    );
+
+    await user.click(
+      await screen.findByRole('textbox', {
+        name: /select resource group/i,
+      })
+    );
+    const groups = screen.getAllByLabelText(/^Resource group/);
+    expect(groups).toHaveLength(2);
+    await user.click(
+      await screen.findByLabelText('Resource group myResourceGroup1')
+    );
+
+    expect(nextButton).not.toHaveClass('pf-m-disabled');
+  }, 10000);
+
+  test('handles change of selected Source', async () => {
+    await setUp();
+
+    const sourceDropdown = await getSourceDropdown();
+
+    await user.click(sourceDropdown);
+    await user.click(
+      await screen.findByRole('option', {
+        name: /azureSource1/i,
+      })
+    );
+
+    await waitFor(() =>
+      expect(
+        screen.getByRole('textbox', {
+          name: /azure tenant guid/i,
+        })
+      ).not.toHaveValue('')
+    );
+
+    await user.click(sourceDropdown);
+    await user.click(
+      await screen.findByRole('option', {
+        name: /azureSource2/i,
+      })
+    );
+    await waitFor(() => {
+      expect(
+        screen.getByRole('textbox', {
+          name: /azure tenant guid/i,
+        })
+      ).toHaveValue('73d5694c-7a28-417e-9fca-55840084f508');
+    });
+
+    await user.click(
+      await screen.findByRole('textbox', {
+        name: /select resource group/i,
+      })
+    );
+    const groups = await screen.findByLabelText(/^Resource group/);
+    expect(groups).toBeInTheDocument();
+    expect(
+      await screen.findByLabelText('Resource group theirGroup2')
+    ).toBeVisible();
+  });
+
+  // test('component renders error state correctly', async () => {
+  //   server.use(
+  //     rest.get(`${PROVISIONING_API}/sources`, (req, res, ctx) =>
+  //      res(ctx.status(500))
+  //     )
+  //   );
+
+  //   await setUp();
+
+  //  await screen.findByText(
+  //   /Sources cannot be reached, try again later or enter an account info for upload manually\./i
+  // );
+  //});
+
+  // set test timeout to 15 seconds
```

Comment:
I wonder why this test doesn't pass. The logic should be implemented, right?

Reply:
Looked into it a bit - the problem lies in closing up `isError` inside of a `isSuccess` if in `AzureSourcesSelect.tsx`

You can test if the error gets rendered by breaking a handler and running msw - in `handlers.js` change
```typescript
  rest.get(`${PROVISIONING_API}/sources`, (req, res, ctx) => {
    const provider = req.url.searchParams.get('provider');
    return res(ctx.status(200), ctx.json(mockSourcesByProvider(provider)));
  }),
```
to
```typescript
  rest.get(`${PROVISIONING_API}/souces`, (req, res, ctx) => {
    const provider = req.url.searchParams.get('provider');
    return res(ctx.status(200), ctx.json(mockSourcesByProvider(provider)));
  }),
```
or something similar (just changing the url to a non-existent one will do the trick).

Then in msw browser the error should get rendered.